packages/components/src/InputTime/InputTime.rebuilt.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { ChangeEvent, useId, useRef } from "react";
+import { useTimePredict } from "./hooks/useTimePredict";
+import { InputTimeProps, InputTimeRebuiltProps } from "./InputTime.Types";
+import { dateToTimeString, timeStringToDate } from "./utils/input-time-utils";
+import { FormFieldWrapper, useFormFieldWrapperStyles } from "../FormField";
+
+export function InputTimeRebuilt({
+  defaultValue,
+  value,
+  onChange,
+  ...params
+}: InputTimeRebuiltProps) {
+  const ref =
+    (params.inputRef as React.RefObject<HTMLInputElement>) ??
+    useRef<HTMLInputElement>(null);
+  const { setTypedTime } = useTimePredict({
+    value,
+    handleChange,
+  });
+
+  const { inputStyle } = useFormFieldWrapperStyles(params);
+
+  const id = getId(params);
+
+  const valueProps = {
+    ...(defaultValue && { defaultValue: dateToTimeString(defaultValue) }),
+    ...(!defaultValue && { value: dateToTimeString(value) }),
+  };
+
+  return (
+    <FormFieldWrapper
+      disabled={params.disabled}
+      size={params.size}
+      align={params.align}
+      inline={params.inline}
+      name={params.name}
+      error={""}
+      identifier={id}
+      descriptionIdentifier={`descriptionUUID--${id}`}
+      invalid={Boolean(params.invalid)}
+      description={params.description}
+      clearable={params.clearable ?? "never"}
+      onClear={handleClear}
+      placeholder={params.placeholder}
+      value={value}
+    >
+      <input
+        ref={ref}
+        type="time"
+        name={params.name}
+        className={inputStyle}
+        onBlur={handleBlur}
+        onChange={handleChangeEvent}
+        onFocus={handleFocus}
+        data-testid="ATL-InputTime-input"
+        onKeyUp={e => {
+          !isNaN(parseInt(e.key, 10)) && setTypedTime(prev => prev + e.key);
+        }}
+        {...valueProps}
+      />
+    </FormFieldWrapper>
+  );
+
+  function handleChangeEvent(event: ChangeEvent<HTMLInputElement>) {
+    handleChange(event.target.value);
+  }
+
+  function handleChange(newValue: string) {
+    onChange?.(timeStringToDate(newValue));
+  }
+
+  function handleBlur(event?: React.FocusEvent<HTMLInputElement>) {
+    params.onBlur?.(event);
+
+    if (ref.current) {
+      if (!ref.current.checkValidity()) {
+        // don't we need to update the value too?
+        // is it valid for them to be out of sync?
+        // onChange?.(undefined);
```

Comment:
I would argue the value in the html input and the value the consumers are aware of should be the same especially since this behaviour was likely handled because RHF was integrated before

Reply:
>After testing it looks like the value of the original implementation wasn't provided to the consumer when it is cleared


hold on, how did you test this???

I still don't see how you'd ever hit this block of code. as far as I can tell it is inaccessible.

Reply:
right but that's the thing, how is this comparison ever going to work?

it seems to me we're always going to be comparing apples and oranges
https://github.com/GetJobber/atlantis/pull/2477#discussion_r2036272865

we don't necessarily **have to** solve this, since it's existing code - but also I'd rather not port over what is basically not functional code if we know it's not going to do anything.

Reply:
ohh, yeah. I meant to say "we can leave this as is"

Reply:
sounds good. I'll make a ticket with a couple of the bugs and inaccessible code paths I've found in the existing implementation but not fuss with that for now.